sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpConnectionScope.cs

```diff
@@ -648,13 +649,80 @@ public override void Dispose()
             }
         }
 
+        private async Task<AmqpSession> CreateSessionIfNeededAsync(AmqpConnection connection, TimeSpan timeout, string transactionGroupKey)
+        {
+            AmqpSession session;
+            FaultTolerantAmqpObject<Controller> faultTolerantController = null;
+            if (transactionGroupKey != null)
+            {
+                if (TransactionGroups.ContainsKey(transactionGroupKey))
+                {
+                    return await TransactionGroups[transactionGroupKey].Session.GetOrCreateAsync(timeout).ConfigureAwait(false);
+                }
+                else
+                {
+                    // each transaction group will have its own session and controller
+                    var faultTolerantSession = new FaultTolerantAmqpObject<AmqpSession>(
+                        async (timeout) =>
+                        {
+                            var stopWatch = ValueStopwatch.StartNew();
+                            AmqpSession session = await CreateAndOpenSessionAsync(connection, timeout.CalculateRemaining(stopWatch.GetElapsedTime())).ConfigureAwait(false);
+                            await CreateControllerAsync(session, timeout.CalculateRemaining(stopWatch.GetElapsedTime())).ConfigureAwait(false);
+                            return session;
+                        },
+                        async (session) => await session.CloseAsync(timeout).ConfigureAwait(false));
+
+                    faultTolerantController = new FaultTolerantAmqpObject<Controller>(
+                        async timeout =>
+                        {
+                            var stopWatch = ValueStopwatch.StartNew();
+                            AmqpSession session = await faultTolerantSession.GetOrCreateAsync(timeout.CalculateRemaining(stopWatch.GetElapsedTime())).ConfigureAwait(false);
+                            return await CreateControllerAsync(session, timeout.CalculateRemaining(stopWatch.GetElapsedTime())).ConfigureAwait(false);
+                        },
+                        controller => controller.Close());
+
+                    var transactionGroup = new AmqpTransactionGroup(faultTolerantSession, faultTolerantController);
+                    TransactionGroups.AddOrUpdate(transactionGroupKey, transactionGroup, (key, old) => transactionGroup);
+
+                    session = await faultTolerantSession.GetOrCreateAsync(timeout).ConfigureAwait(false);
+
+                    // When using transaction groups, the controller needs to be opened before the link is established, i.e. we can't
+                    // wait until a transaction is declared to open the controller.
+                    await faultTolerantController.GetOrCreateAsync(timeout).ConfigureAwait(false);
```

Comment:
I'm surprised this didn't get flagged by the analyzer for "don't ignore return values", which I seem to trip in these scenarios.   You may want to consider an explicit discard to guard against that popping up in the future.  
```suggestion
                    _ = await faultTolerantController.GetOrCreateAsync(timeout).ConfigureAwait(false);
```